core/src/main/java/org/jivesoftware/sparkimpl/certificates/CertificateController.java

```diff
@@ -206,67 +204,107 @@ private void addRowsToTableModel(List<CertificateModel> certList, Object[] certE
+        if (getAliasKeyStorePath(alias).equals(TRUSTED)) {
 
-            if (checked) {
-
-                try {
-                    moveCertificate(TRUSTED, EXCEPTIONS);
-                } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException ex) {
-                    Log.error("Error at moving certificate from trusted list to the exception list", ex);
-                }
+            if(checked) {
+                addCertToExceptions(alias);
             } else {
-                try {
-                    moveCertificate(EXCEPTIONS, TRUSTED);
-                } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException ex) {
-                    Log.error("Error at moving certificate from exceptions list to trusted list", ex);
-                }
+                removeCertFromExceptions(alias);
             }
-        } else if (getAliasKeyStorePath(alias).equals(DISPLAYED_CACERTS)
-                || getAliasKeyStorePath(alias).equals(CACERTS_EXCEPTIONS)) {
-            if (checked) {
-
-                try {
-                    moveCertificate(DISPLAYED_CACERTS, CACERTS_EXCEPTIONS);
-                } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException ex) {
-                    Log.error("Error at moving certificate from trusted list to the exception list", ex);
-                }
+            
+        }else if (getAliasKeyStorePath(alias).equals(DISPLAYED_CACERTS)) {
+            if(checked) {
+                addCaCertToExceptions(alias);
             } else {
-                try {
-                    moveCertificate(CACERTS_EXCEPTIONS, DISPLAYED_CACERTS);
-                } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException | IOException ex) {
-                    Log.error("Error at moving certificate from exceptions list to trusted list", ex);
-                }
+                removeCaCertFromExceptions(alias);
             }
         }
     }
-	
-	/**
-	 * Return information if certificate is on exception list.
-	 * 
-	 * @param Certificate Model entry
-	 */
+
+    /**
+     * Return information if certificate is on exception list.
+     * 
+     * @param Certificate
+     *            Model entry
+     */
     @Override
-	public boolean isOnExceptionList(CertificateModel cert) {
-		if(exemptedCertificates.contains(cert)){
-		    return true;
-		}else if (exemptedCacerts.contains(cert)){
-		    return true;
-		}else {
-		    return false;
-		}
+    public boolean isOnExceptionList(CertificateModel cert) {
+        try {
+            if (exceptionsStore.getCertificateAlias(cert.getCertificate()) != null) {
+                return true;
+            } else if (exceptionsCaStore.getCertificateAlias(cert.getCertificate()) != null) {
+                return true;
+            } else {
+                return false;
+            }
+        } catch (KeyStoreException e) {
+            return false;
```

Comment:
If an error occurs, we shouldn't assume 'yes' or 'no'. Just throw or show the error instead.